apps/next-evans/pages/friends.tsx

```diff
@@ -0,0 +1,187 @@
+import React, { useState } from 'react';
+import { PrismaClient } from '@prisma/client';
+import invariant from 'tiny-invariant';
+import type { GetStaticProps } from 'next';
+import type { friends as Friend } from '@prisma/client';
+import { LANG } from '@/lib/env';
+import FriendsPageHero from '@/components/pages/friends/FriendsPageHero';
+import { t } from '@/../../libs-ts/locale/src';
+import FriendCard from '@/components/pages/friends/FriendCard';
+import ControlsBlock from '@/components/pages/friends/ControlsBlock';
+import CompilationsBlock from '@/components/pages/friends/CompilationsBlock';
+
+export const getStaticProps: GetStaticProps<Props> = async () => {
+  const client = new PrismaClient();
+  const friends = await client.friends.findMany({
+    where: { lang: LANG },
+    select: {
+      name: true,
+      created_at: true,
+      gender: true,
+      slug: true,
+      born: true,
+      died: true,
+      friend_residences: {
+        select: {
+          city: true,
+          region: true,
+        },
+      },
+      documents: {
+        select: {
+          id: true,
+          editions: {
+            select: {
+              is_draft: true,
+            },
+          },
+        },
+      },
+    },
+  });
+
+  return {
+    props: {
+      friends: friends
+        .filter((friend) => friend.friend_residences[0] !== undefined)
```

Comment:
i think you should remove this line. the invariant below will break the build if a friend doesn't have a residence, which we want, they should always have one